interview-questions.md

```diff
@@ -8,42 +8,54 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: TLDR, in Ruby, integers refer to numbers that are whole, or do not have a decimal point, examples of an integer would be: 1,2,-1,-2, whereas floats are number data types that contain decimial points; examples of a float would be 1.2, 3.9, -2.3 and so forth. Other key differences to know is that precision-wise between the two, floats have lower precision when compared to integers due to the decimal part of the float sometime being very long.
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: I am unsure what implicit return refers to. If I were to guess, I believe it refers to certain code being brought back automatically in the case of classes?
 
-Researched answer:
+Researched answer: Implicit return in Ruby refers to how a block will automatically return the value of the last code line. This is why it is not necessary to use "return" in Ruby as you would have to in JavaScript, although you can if you would like, but it does not make a difference.
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: A block in Ruby refers to methods that have anoynmous functions pass down additional information.
 
-Researched answer:
+Researched answer: Blocks, in Ruby is a way that you are able to group multiple statements together and providing them as a unit of code to be executed. They can be used for iteration, defining the method, and exception handling. Similiarly, they are almost the same as a method, and the only thing causing them to be different is that with blocks, they do not belong to an object.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: In order to extract a value in a Ruby hash, you must first call the name of the hash, followed by in brackets[] and then within your brackets, use your symbol(:) followed by the name of value you would want to call that you previously created. An example of how this would look would be:
 
-Researched answer:
+caffeine_amount[:coffee] = '95mg'
+caffeine_amount[:tea] = '40mg' 
+caffeine_amount[:energy_drink] = '300mg'
+
+If you want to extract the value of energy drink you would:
+
+p caffeine_amount[:energy_drink]
+output: "300mg"
+
+Researched answer: To extract a value in a Ruby hash, you can use the .fetch method or using the [] operands. The square bracket operators[] works as stated above, however with the .fetch method in order to use that, you would simply just follow it with the name of your hash, but instead use parantheses() followed by a symbol: and then the name of the value. If the key is note found within the hash, it would return a "KeyError" when you go to print in terminal. An example of what this would then look like is:
+
+p caffeine_amount.fetch(:coffee)
+output: "95mg"
```

Comment:
quality answers. Now that you have placed it in your own words, I recommend going back through and ensuring you are using the web development vocabulary. The symbol is the colon and variable used for the key.

For example:
`To extract a value in a Ruby hash, you can use the .fetch method or using the [] operands. The square bracket operators[] works as stated above, however with the .fetch method in order to use that, you would simply just follow it with the name of your hash, but instead use parantheses() followed by a symbol: and then the name of the value. If the key is note found within the hash, it would return a "KeyError" when you go to print in terminal.`

Using vocabulary
`To extract a value in a Ruby hash, you can reference its symbol using the .fetch method or bracket notation. The bracket notation works as stated above, however with the .fetch method in order to use that, you would simply just the symbol (key) as an argument. If the symbol (key) is not found within the hash, a "KeyError"  would be returned in the terminal.`